ext/MitsubaLoader/CMitsubaMaterialCompilerFrontend.cpp

```diff
@@ -0,0 +1,597 @@
+                if (node_specular->ndf == IR::CMicrofacetSpecularBSDFNode::ENDF_ASHIKHMIN_SHIRLEY)
+                    getFloatOrTexture(current->plastic.alphaV, node_specular->alpha_v);
+                else
+                    node_specular->alpha_v = node_specular->alpha_u;
+            }
+            else
+            {
+                node_specular->setSmooth();
+            }
+            auto* node_diffuse = static_cast<IR::CMicrofacetDiffuseBSDFNode*>(diffuse);
+            getSpectrumOrTexture(current->plastic.diffuseReflectance, node_diffuse->reflectance);
+            node_diffuse->alpha_u = node_specular->alpha_u;
+            node_diffuse->alpha_v = node_specular->alpha_v;
+            node_diffuse->eta = node_specular->eta;
+        }
+        break;
+        case CElementBSDF::DIELECTRIC:
+        case CElementBSDF::THINDIELECTRIC:
+        case CElementBSDF::ROUGHDIELECTRIC:
+        {
+            nextSym = ir->allocNode<IR::CBSDFCombinerNode>(IR::CBSDFCombinerNode::ET_FRESNEL_BLEND);
+            nextSym->children.count = 2u;
+
+            const float eta = current->dielectric.intIOR/current->dielectric.extIOR;
+            _IRR_DEBUG_BREAK_IF(eta==1.f);
+            if (eta==1.f)
+                os::Printer::log("WARNING: Dielectric with IoR=1.0!", current->id, ELL_ERROR);
+
+            auto& refl = nextSym->children[0];
+            refl = ir->allocNode<IR::CMicrofacetSpecularBSDFNode>();
+            auto& trans = nextSym->children[1];
+            trans = ir->allocNode<IR::CMicrofacetSpecularBSDFNode>();
+
+            auto* node_refl = static_cast<IR::CMicrofacetSpecularBSDFNode*>(refl);
+            auto* node_trans = static_cast<IR::CMicrofacetSpecularBSDFNode*>(trans);
+
+            node_refl->scatteringMode = IR::CMicrofacetSpecularBSDFNode::ESM_REFLECT;
+            node_refl->shadowing = IR::CMicrofacetSpecularBSDFNode::EST_SMITH;
+            node_refl->eta = IR::INode::color_t(eta);
+            if (currType == CElementBSDF::ROUGHDIELECTRIC)
+            {
+                node_refl->ndf = ndfMap[current->dielectric.distribution];
+                getFloatOrTexture(current->dielectric.alphaU, node_refl->alpha_u);
+                if (node_refl->ndf == IR::CMicrofacetSpecularBSDFNode::ENDF_ASHIKHMIN_SHIRLEY)
+                    getFloatOrTexture(current->dielectric.alphaV, node_refl->alpha_v);
+                else
+                    node_refl->alpha_v = node_refl->alpha_u;
+            }
+            else
+            {
+                node_refl->setSmooth();
+            }
+            node_trans->scatteringMode = IR::CMicrofacetSpecularBSDFNode::ESM_TRANSMIT;
+            node_trans->ndf = node_refl->ndf;
+            node_trans->shadowing = node_refl->shadowing;
+            node_trans->alpha_u = node_refl->alpha_u;
+            node_trans->alpha_v = node_refl->alpha_v;
+            node_trans->eta = node_refl->eta;
+
+            thin = (currType == CElementBSDF::THINDIELECTRIC);
```

Comment:
`thin` seems to have the same issue as `twosided`

Reply:
which is only the rest of the branch from thin onwards should be thin, not the whole AST